scripts/deb/post-install.sh

```diff
@@ -51,6 +51,16 @@ test -d $LOG_DIR || mkdir -p $LOG_DIR
 chown -R -L telegraf:telegraf $LOG_DIR
 chmod 755 $LOG_DIR
 
+STATE_DIR=/var/lib/telegraf
+test -d $STATE_DIR || mkdir -p $STATE_DIR
+chown -R -L root:telegraf $STATE_DIR
+chmod 770 $STATE_DIR
```

Comment:
There two things with this:
1. I don't think we want to use `chown -R $STATE_DIR` since on upgrades, if the user put other files in this directory, it would also be chowned, which is an unexpected side effect
2. I found the use of `-L` surprising. Why are we trying to follow symlinks on a directory we just created that we didn't create as a symlink?
3. since this looks like it will `chown` the STATE_DIR on package upgrades, it will overwrite system permissions that the user may have put in place

I wonder it it would be better to (note best practice of also quoting the shell variables (this is a nitpick since nothing here is user controllable; it just has better style)::
```
STATE_DIR=/var/lib/telegraf
test -d "$STATE_DIR" || {
    mkdir -m 0770 -p "$STATE_DIR"
    chown root:telegraf "$STATE_DIR"
}
```

Reply:
> Why are we trying to follow symlinks on a directory we just created that we didn't create as a symlink?

heh I copied this from the log folder creation directly above this

I like your suggestion and will adapt both to match.